infrastructure/controller/pkg/endpoints/components_install_uninstall.go

```diff
@@ -0,0 +1,133 @@
+package endpoints
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/airyhq/airy/lib/go/payloads"
+	helmCli "github.com/mittwald/go-helm-client"
+	"helm.sh/helm/v3/pkg/repo"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+)
+
+type ComponentsInstallUninstall struct {
+	cli       helmCli.Client
+	namespace string
+}
+
+func MustNewComponentsInstallUninstall(namespace string, kubeConfig *rest.Config, reposFilePath string) ComponentsInstallUninstall {
+	cli, err := helmCli.NewClientFromRestConf(&helmCli.RestConfClientOptions{
+		Options: &helmCli.Options{
+			Namespace: namespace,
+		},
+		RestConfig: kubeConfig,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	repos, err := getReposFromFile(reposFilePath)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, r := range repos {
+		if r.Password != "" {
+			r.PassCredentialsAll = true
+		}
+
+		if err := cli.AddOrUpdateChartRepo(r); err != nil {
+			log.Fatal(err)
+		}
+		klog.Info("Added ", r.Name, " repo")
+	}
+
+	return ComponentsInstallUninstall{namespace: namespace, cli: cli}
+}
+
+func (s *ComponentsInstallUninstall) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintf(w, `{"error": "not able to read body"}`)
+		return
+	}
+
+	chartName, releaseName, err := getChartNameFromBlob(body)
+	if err != nil {
+		klog.Error(err.Error())
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintf(w, `{"error": "invalid component name"}`)
+		return
+	}
+
+	chartSpec := &helmCli.ChartSpec{
+		ReleaseName: releaseName,
+		ChartName:   chartName,
+		Namespace:   s.namespace,
+		UpgradeCRDs: true,
+		Replace:     true,
```

Comment:
I guess this is only needed because the job is `async` and we should be able to install the same chart multiple times. Otherwise, I think it is better to error if the chart is already installed.